pkg/skaffold/runner/runcontext/v2/context.go

```diff
@@ -291,11 +291,22 @@ func (rc *RunContext) DigestSource() string {
 	return constants.RemoteDigestSource
 }
 
+func setPipelineConfigName(pipeline *latest.Pipeline, configName string) {
+	pipeline.ConfigName = configName
+
+	if len(pipeline.ConfigName) == 0 {
+		configNameUuid, _ := uuid.NewUUID()
+		pipeline.ConfigName = configNameUuid.String()
+	}
+}
+
 func GetRunContext(ctx context.Context, opts config.SkaffoldOptions, configs []schemaUtil.VersionedConfig) (*RunContext, error) {
 	var pipelines []latest.Pipeline
 	for _, cfg := range configs {
 		if cfg != nil {
-			pipelines = append(pipelines, cfg.(*latest.SkaffoldConfig).Pipeline)
+			pipeline := cfg.(*latest.SkaffoldConfig).Pipeline
```

Comment:
can you instead handle the config names in the `Pipelines` struct:
```diff
// Pipelines encapsulates multiple config pipelines
type Pipelines struct {
	pipelines            []latest.Pipeline
+	pipelinesByConfig    map[string]latest.Pipeline
	pipelinesByImageName map[string]latest.Pipeline
}

```
and initialize it this way in `GetRunContext`:

```go
var pipelines map[string]latest.Pipeline
for _, vcfg := range configs {
	if vcfg != nil {
        cfg := vcfg.(*latest.SkaffoldConfig)
        cfgName := cfg.Metadata.Name
        if len(cfgName) == 0 {
            randomName, _ := uuid.NewUUID()
            cfgName = randomName.String()
        }
        pipelines[cfgName] = cfg.Pipeline
    }
}
```
Then we can support both methods:
```go
func (rc *RunContext) GetPipelines()
```
and
```go
func (rc *RunContext) GetPipelineForConfig(configName string)
```

Reply:
Done!